src/rule-creator.ts

```diff
@@ -3,15 +3,20 @@ import { ESLintUtils } from '@typescript-eslint/experimental-utils'
 import type { NGRX_MODULE } from './utils'
 import { docsUrl, ngrxVersionSatisfies, NGRX_MODULE_PATHS } from './utils'
 
-export type NgRxRuleModule<TMessageIds extends string = string> =
-  TSESLint.RuleModule<TMessageIds, unknown[]> & {
-    meta: { ngrxModule: NGRX_MODULE; version?: string }
-  }
-
-type CreateRuleMetaDocs = Omit<TSESLint.RuleMetaDataDocs, 'url'>
+type Meta<TMessageIds extends string> =
+  | TSESLint.RuleMetaData<TMessageIds> & {
+      ngrxModule?: NGRX_MODULE
+      version?: string
+    }
 type CreateRuleMeta<TMessageIds extends string> = {
-  docs: CreateRuleMetaDocs
-} & Omit<NgRxRuleModule<TMessageIds>['meta'], 'docs'>
+  docs: Omit<TSESLint.RuleMetaDataDocs, 'url'>
+} & Omit<Meta<TMessageIds>, 'docs'>
+export type NgRxRuleModule<
+  TOptions extends readonly unknown[],
+  TMessageIds extends string,
+> = Omit<TSESLint.RuleModule<TMessageIds, TOptions>, 'meta'> & {
```

Comment:
Fun fact: most of the interfaces/types from `@typescript-eslint` requires this generic order: `<TMessageIds, TOptions>`, however specifically the `ESLintUtils.RuleCreator` requires a different order `<TOptions, TMessageIds>`